### ISSUE TITLE
cancel_order handler doesn't check nonce

### ISSUE THREAD

**Issue by nicoladj77** [woocommerce/woocommerce#16875]
Hi,

it looks like the [cancel_order handler](https://github.com/woocommerce/woocommerce/blob/master/includes/class-wc-form-handler.php#L674) doesn't check for the nonce validity, which should be checked

**Comment by nicoladj77** [comment]
Something like

```
		if (
			isset( $_GET['unskip_order'] ) &&
			isset( $_GET['order'] ) &&
			isset( $_GET['order_id'] ) &&
			( isset( $_GET['_wpnonce'] ) && wp_verify_nonce( $_GET['_wpnonce'], 'woocommerce-cancel_order' ) )
		) {

               }
```

**Comment by nicoladj77** [comment]
Closing this in favor of https://github.com/woocommerce/woocommerce/pull/16877

**Comment by claudiulodro** [comment]
Fixed in #16877
